pennylane_snowflurry/utility/graph.py

```diff
@@ -0,0 +1,185 @@
+    for mono in vf2.subgraph_monomorphisms_iter():
+       return {v : k for k, v in mono.items()}
+    return None
+
+def find_largest_subgraph_isomorphism_vf2(circuit : nx.Graph, machine : nx.Graph):
+    """
+    Uses vf2 and combinations to find the largest common graph between two graphs
+    """
+    edges = [e for e in circuit.edges]
+    for i in reversed(range(len(edges) + 1)):
+        for comb in combinations(edges, i):
+            result = _find_isomorphisms(nx.Graph(comb), machine)
+            if result: return result
+
+def find_largest_subgraph_isomorphism_imags(circuit : nx.Graph, machine : nx.Graph):
+    """
+    Uses IMAGS to find the largest common graph between two graphs
+    """
+    ismags = ISMAGS(machine, circuit)
+    for mapping in ismags.largest_common_subgraph():
+        return {v:k for (k, v) in mapping.items()} if mapping is not None and len(mapping) > 0 else mapping
+
+def shortest_path(a : int, b : int, graph : nx.Graph, excluding : list[int] = [], prioritized_nodes : list[int] = [], use_benchmark=True):
+    """
+    find the shortest path between node a and b
+
+    Args :
+        a : start node
+        b : end node
+        graph : the graph to find a path in
+        excluding : nodes we dont want to use
+        prioritized_nodes : nodes we want to use if possible
+    """
+    r1_cz_fidelities = get_readout1_and_cz_fidelities() if use_benchmark else {}
+    g_copy = deepcopy(graph)
+    g_copy.remove_nodes_from(excluding)
+
+    def weight(node_u, node_v):
+        if not use_benchmark: 
+            return 1
+    
+        weights = [v for k, v in r1_cz_fidelities[keys.czGateFidelity].items() if node_u in k and node_v in k]
+        r1_node_u = r1_cz_fidelities[keys.readoutState1Fidelity][str(node_u)]
+        r1_node_v = r1_cz_fidelities[keys.readoutState1Fidelity][str(node_v)]
+        
+        if len(weights) < 1:
+            return 10
+        w = 4 - weights[0] - r1_node_u - r1_node_v
+        
+        if node_u in prioritized_nodes or node_v in prioritized_nodes:
+            return w - 1
+        return w
+    
+    return nx.astar_path(g_copy, a, b, weight = lambda u, v, _: weight(u, v))
+
+def find_best_neighbour(wire, topology : nx.Graph, use_benchmark = True):
+    """
+    
+    """
+    neigh = list(topology.neighbors(wire))
+    return max(neigh, key = lambda n : calculate_score(n, topology, use_benchmark))
+        
+
+def find_best_wire(graph : nx.Graph, excluded : list[int] = [], use_benchmark = True):
+    """
+    find node with highest degree in graph
+    """
+    g = deepcopy(graph)
+    g.remove_nodes_from(excluded)
+    return max([n for n in g.nodes], key=lambda n: calculate_score(n, g, use_benchmark))
+
+def find_closest_wire(a : int, machine_graph : nx.Graph, excluding : list[int] = [], prioritized : list[int] = [], use_benchmark = True):
+    """
+    find node in graph that is closest to given node, not considering arbitrary excluding list
```

Comment:
ajouter les arguments dans le docstring

Clarifier le "not considering exluding list"